src/peft/tuners/boft/layer.py

```diff
@@ -315,11 +315,11 @@ def update_layer(
 
             boft_block_num = int(self.in_features // boft_block_size)
 
+        elif boft_block_size != 0 and boft_block_num != 0:
+            raise ValueError(f"You can only specify either boft_block_size ({boft_block_size}) or boft_block_num ({boft_block_num}), but not both simultaneously.")
+
         else:
-            raise ValueError(
-                f"You can only specify either boft_block_size ({boft_block_size}) or boft_block_num ({boft_block_num}), but not both simultaneously or setting both"
-                "to be 0, because boft_block_size x boft_block_num != in_features."
-            )
+            raise ValueError(f"Either `boft_block_size` or `boft_block_num` must be non-zero. Currently, boft_block_size = {boft_block_size} and boft_block_num = {boft_block_num}.")
```

Comment:
Do we need these checks? We already have checks here:

https://github.com/huggingface/peft/blob/41c274ecac6247a63e3b10f8536904f3ead82213/src/peft/tuners/boft/config.py#L127-L133

Reply:
Yes, you are right, I forgot there is already a check in the config.py Updated.